source/HL_i2c.c

```diff
@@ -140,18 +141,18 @@ void i2cInit(void)
                  | (uint32)(1U);     /* scl pin */
 
     /** - set interrupt enable */
-    i2cREG1->IMR    = (uint32)((uint32)0U << 6U)     /* Address as slave interrupt      */
-                    | (uint32)((uint32)0U << 5U)     /* Stop Condition detect interrupt */
-                    | (uint32)((uint32)0U << 4U)     /* Transmit data ready interrupt   */
-                    | (uint32)((uint32)0U << 3U)     /* Receive data ready interrupt    */
-                    | (uint32)((uint32)0U << 2U)     /* Register Access ready interrupt */
-                    | (uint32)((uint32)0U << 1U)     /* No Acknowledgment interrupt    */
-                    | (uint32)((uint32)0U);     /* Arbitration Lost interrupt      */
+    i2cREG1->IMR    = (uint32)((uint32)1U << 6U)     /* Address as slave interrupt      */

```

Comment:
If I understand this correctly, are we sure we want all these interrupts enabled? The NoACK intr was discussed in the Tues advisor meeting, for example, and I can see it, but the Address as slave, Stop Condition, Register Access, and Arbitration Lost interrupts are less obvious to me.

Reply:
Oh, and here and below, perhaps I am commenting on generated code, so the comments may not be totally appropriate...

Reply:
I think I could disable the address as slave interrupt. But then I don't know what the behavior will be if the flag gets set in error. My IRQ code explicitly unsets it just in case

Reply:
Stop condition is used in every case that a transfer is complete, whether in error or nominally. So it is necessary to give the semaphore letting the calling thread know it is done

Arbitration lost is set so if there's an error where the bus is held low there will be a signal to attempt to recover

Register access seems to be only fired when there's a NACK in the start byte. I think it's supposed to be fired every time a register is ready, as in the TX/RX interrupt for example, but is a lower priority so it doesn't actually fire, then the routines for the other interrupts unset this bit as part of their regular actions. The technical reference for the module shows the ways this interrupt can be cleared

Reply:
So, to summarize, if it's not broke...

That said, Register Access seems redundant the way you describe it, so maybe worth losing? Without digging into the data sheet myself in detail, I tend to not enable things just because they are there. The other stuff seems okay.

Your choice on this, just be sure you are comfortable.